concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -1196,15 +1197,69 @@ handleContractUpdateV1 originAddr istance checkAndGetSender transferAmount recei
                                         -- Charge for updating the pointer in the instance to the new module.
                                         tickEnergy Cost.initializeContractInstanceCreateCost
                                         go (resumeEvent True:upgradeEvent:interruptEvent:events) =<< runInterpreter (return . WasmV1.resumeReceiveFun rrdInterruptedConfig rrdCurrentState False newBalance WasmV1.Success Nothing)
-
+                  WasmV1.QueryAccountBalance {..}  -> do
+                    newBalance <- getCurrentContractAmount Wasm.SV1 istance
+                    -- TODO Charge query cost or maybe not
```

Comment:
What is the status of this todo?

Reply:
I have resolved the todos, by adding a cost to each of the queries as according to the comment by ales.